PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/spark/common2/SparkSubmitHelper.java

```diff
@@ -179,6 +179,8 @@ public void printRunningLogStreamingly(/*Project project,*/ int id, IClusterDeta
 
             if (isFailedJob) {
                 postEventProperty.put("IsRunningSucceed", "false");
+                HttpResponse httpResponse = SparkBatchSubmission.getInstance().getBatchJobFullLog(clusterDetail.getConnectionUrl() + "/livy/batches", id);
+                postEventProperty.put("SubmitFailedReason", httpResponse.getMessage().substring(0, 50));
```

Comment:
Fixed

---
In reply to: [120869247](https://github.com/Microsoft/azure-tools-for-java/pull/603#discussion_r120869247) [](ancestors = 120869247)